[mlir][linalg][nfc] Move vectorization tests

banach-space opened the pull request:
Moves all the remaining Linalg vectorization tests from:
  * `mlir/tests/Dialect/Linalg/*`

to:
  * `mlir/tests/Dialect/Linalg/vectorization/*`

To maintain consistency within tests,  `vectorize-convolution.mlir` 
was updated to use:
  *  `transform.structured.vectorize_children_and_apply_patterns` 

instead of:
  * `-test-linalg-transform-patterns=test-linalg-to-vector-patterns`

This change required minor updates to some `CHECK` lines, reflecting
only reordering of ops due to an additional pattern being applied.

Closes #141025

banach-space committed:
[mlir][linalg][nfc] Move vectorization tests

Moves:
  * mlir/test/Dialect/Linalg/vectorization-with-patterns.mlir
  * mlir/test/Dialect/Linalg/vectorization.mlir
  * mlir/test/Dialect/Linalg/vectorization-unsupported.mlir

to:
  * mlir/test/Dialect/Linalg/vectorization/

This patch implements part of #141025 - please see the ticket for full
context.

hanhanW reviewed (commented):
I'm okay with reviewing a large change if you break it into several commits, like what you're doing now. (I think the only difference is that it is PR-based, and I can do commit-based within a PR.)

Would it be better to submit a single PR for the ticket?

banach-space committed:
fixup! [mlir][linalg][nfc] Move vectorization tests

* In vectorize-convolution.mlir, switch from
  `-test-linalg-transform-patterns=test-linalg-to-vector-patterns` to
  `transform.structured.vectorize_children_and_apply_patterns` (for
  consistency with other tests for vectorization).
* Move + rename the remaining vectorization tests.
* Add comments.

hanhanW commented:
You may want to add `Closes https://github.com/llvm/llvm-project/issues/141025` in the PR description. This is the last PR, right?
